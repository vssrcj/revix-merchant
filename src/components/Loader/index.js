/**
 *
 * Loader
 *
 */

import React from 'react';
import { Spin } from 'antd';
import PropTypes from 'prop-types';
// import styled from 'styled-components';

function Loader(props) {
  return (
    <Spin
      delay={props.delay}
      indicator={props.indicator}
      size={props.size}
      spinning={props.spinning}
      tip={props.tip}
      wrapperClassName={props.wrapperClassName}
    />
  );
}

Loader.propTypes = {
  delay: PropTypes.number,
  indicator: PropTypes.node,
  size: PropTypes.string,
  spinning: PropTypes.bool,
  tip: PropTypes.string,
  wrapperClassName: PropTypes.string,
};

export default Loader;
