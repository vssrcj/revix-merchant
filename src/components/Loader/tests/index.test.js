/**
 *
 * Tests for Loader
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Loader from '../index';

describe('<Loader />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Loader />);
    expect(spy).not.toHaveBeenCalled();
  });

  /**
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Loader />);
    expect(firstChild).toMatchSnapshot();
  });
});
