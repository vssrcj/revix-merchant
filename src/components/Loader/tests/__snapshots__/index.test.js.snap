// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Should render and match the snapshot 1`] = `
<div
  class="ant-spin ant-spin-spinning"
>
  <span
    class="ant-spin-dot ant-spin-dot-spin"
  >
    <i
      class="ant-spin-dot-item"
    />
    <i
      class="ant-spin-dot-item"
    />
    <i
      class="ant-spin-dot-item"
    />
    <i
      class="ant-spin-dot-item"
    />
  </span>
</div>
`;
